docs/yarn_classic.md

```diff
@@ -0,0 +1,154 @@
+
+Prefetching dependencies for Yarn in Cachi2 is be done using Yarn's offline mirror feature.
+The project must be configured to use the offline mirror feature. Refer to
+[Setting up offline mirror][] below for details.
+
+
+### Supported dependencies types
+
+Yarn Classic PM is capable of processing the following types of dependencies:
+ * packages from registries;
+ * packages from git repos;
+ * packages from http/https URLs;
+ * packages from local paths;
+ * local workspace packages;
+ * local link packages.
+
+### Dealing with Yarn Zero-Installs
+
+Yarn Classic's [Plug'n'Play][] feature is not supported. Any package that uses
+it will be rejected. For further details please refer to [Yarn v3 documentation][].
+
+## Fetching dependencies for Yarn Classic projects
+
+The process of fetching dependencies for Yarn Classic is similar to that for any other
+package manager. The name of package manager is `yarn_classic`, and it does not expect
+any additional arguments.
+
+Cachi2 ``fetch-deps`` shell command:
+
+```shell
+cachi2 fetch-deps \
+  --source ./my-repo \
+  --output ./cachi2-output \
+  '<JSON input>'
+```
+
+where JSON input is:
+```jsonc
+{
+  // "yarn_classic" tells Cachi2 to process Yarn packages
+  "type": "yarn_classic",
+  // path to the package (relative to the --source directory)
+  // defaults to "."
+  "path": ".",
+}
+```
+
+or more simply by just invoking:
+``cachi2 fetch-deps yarn_classic``
+
+For complete example of how to pre-fetch dependencies, see [Pre-fetch dependencies][].
+
+## Handling of yarn-specific config files
+
+Yarn Classic allows a user to provide additional configuration via [.yarnrc][]
+and [.npmrc][].  **Cachi2 ignores these settings during prefetch phase**.
+However a `.yarnrc` could be used for setting up an offline mirror
+([Setting up offline mirror][]).  These settings will be applied during
+a build phase.
+
+
+### Setting up offline mirror
+
+A project that is to be hermetically built must be configured to use an offline mirror.
+This means that Yarn will store compressed archives on the file system in a mirror directory
+and will install them from there later without network access.
+
+A project must be configured to use offline mirror either by providing a `.yarnrc` file
+or by setting up several environment variables. In case when `.yarnrc` is preferred
+it must contain the following lines:
+
+```ini
+yarn-offline-mirror <absolute path to the request output directory>
+yarn-offline-mirror-pruning false
+```
+
+It can be either directly written to or `yarn config` could be used:
+```bash
+$ yarn config set yarn-offline-mirror <absolute path to the request output directory>
+$ yarn config set yarn-offline-mirror-pruning false
+```
+
+In case when environment variables approach is preferred the following
+variables must be set:
+
+```bash
+YARN_YARN_OFFLINE_MIRROR=<absolute path to the request output directory>
+YARN_YARN_OFFLINE_MIRROR_PRUNING=false
```

Comment:
These environment variables are set by cachi2, aren't they ?
The offline mirror is basically `--output` option.